Source/Core/BoogiePL.atg

```diff
@@ -1217,19 +1217,25 @@ Power<out Expr/*!*/ e0>
 
 /*------------------------------------------------------------------------*/
 UnaryExpression<out Expr/*!*/ e>
-= (. Contract.Ensures(Contract.ValueAtReturn(out e) != null); IToken/*!*/ x;
+= (. Contract.Ensures(Contract.ValueAtReturn(out e) != null); IToken/*!*/ x; IsConstructor isConstructor;
      e = dummyExpr;
   .)
   ( "-"                        (. x = t; .)
     UnaryExpression<out e>     (. e = Expr.Unary(x, UnaryOperator.Opcode.Neg, e); .)
   | NegOp                      (. x = t; .)
     UnaryExpression<out e>     (. e = Expr.Unary(x, UnaryOperator.Opcode.Not, e); .)
-  | CoercionExpression<out e>
+  | CoercionExpression<out e> [IF(la.val == "is") IsConstructor<out x, out isConstructor> (. e = new NAryExpr(x, isConstructor, new List<Expr> { e }); .)]
```

Comment:
This seems wrong to me. The effect is that `is` binds stronger than `!` (and `-`). So,

    !a is node

means

    !(a is node)

which is not what `!a is node` looks like to me. I suggest you undo the change on L1227 and instead do:

```
Power<out Expr/*!*/ e0>
= (.Contract.Ensures(Contract.ValueAtReturn(out e0) != null); IToken/*!*/ x; Expr/*!*/ e1; .)
  IsConstructor<out e0>
  [
    "**"           (. x = t; .)
    /* recurse because exponentation is right-associative */
    Power<out e1>  (. e0 = Expr.Binary(x, BinaryOperator.Opcode.Pow, e0, e1); .)
  ]
  .

IsConstructor<out Expr/*!*/ e0>
= (. Contract.Ensures(Contract.ValueAtReturn(out e0) != null); IToken/*!*/ x, id; .)
  UnaryExpression<out e0>
  [ "is"            (. x = t; .)
    Ident<out id>
    (. var isConstructor = new IsConstructor(id, id.val);
       e0 = new NAryExpr(x, isConstructor, new List<Expr> { e0 });
    .)
  ]
  .
```

That makes the binding power of `is` stronger than the binary `**` and weaker than the unary `!` and `-`. Then, also change `IsConstructor.Emit` in `AbsyExpr.cs` to use:

``` c#
      const int opBindingStrength = 0x68;
```